src/Middleware/RateLimiting/src/RateLimitingApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Extension methods for the RateLimiting middleware.
+/// </summary>
+public static class RateLimitingApplicationBuilderExtensions
+{
+    /// <summary>
+    /// Enables rate limiting for the application.
+    /// </summary>
+    /// <param name="app"></param>
+    /// <returns></returns>
+    public static IApplicationBuilder UseRateLimiter(this IApplicationBuilder app)
+    {
+        if (app == null)
+        {
+            throw new ArgumentNullException(nameof(app));
+        }
+
+        return app.UseMiddleware<RateLimitingMiddleware>();
+    }
+
+    /// <summary>
+    /// Enables rate limiting for the application.
+    /// </summary>
+    /// <param name="app"></param>
+    /// <param name="options"></param>
+    /// <returns></returns>
+    public static IApplicationBuilder UseRateLimiter(this IApplicationBuilder app, RateLimiterOptions options)
+    {
+        if (app == null)
+        {
+            throw new ArgumentNullException(nameof(app));
+        }
+        if (options == null)
```

Comment:
Are these options just being thrown away?

I see we have tests showing that the `RateLimitingMiddleware` throws `ArguementNullException`s, but none for the `UseRateLimiter` methods. We should add some. We should also make sure we're using the options that are passed in over what was configured in the service collection like `UseStaticFiles` does.

Reply:
Whoops, not using the options was an oversight. Will fix & add tests